src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -570,16 +570,22 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
         if (lock) {
             params = params.set('lock', 'true');
         }
-
-        return this.http.get<ProgrammingSubmission>(url, { params });
+        const returnValue = this.http.get<ProgrammingSubmission>(url, { params });
+        return returnValue;
     }
 
     /**
      * Locks the submission of the participation for the user
      * @param participationId
+     * @param correctionRound
      */
-    lockAndGetProgrammingSubmissionParticipation(participationId: number): Observable<Participation> {
-        return this.http.get<Participation>(`api/programming-submissions/${participationId}/lock`);
+    lockAndGetProgrammingSubmissionParticipation(participationId: number, correctionRound = 0): Observable<Participation> {
+        let params = new HttpParams();
+        if (correctionRound !== 0) {
```

Comment:
Maybe this condition is stronger? `correctionRound > 0` since here it's possible to input negative numbers

Reply:
Good catch!